512.md

```diff
@@ -0,0 +1,173 @@
+```
+{
+  "tags":[
+    ["a", "<addr>", "<relay>", "site"], // addr of the target site
+    ["a"/"e", "<addr/id>", <relay>], // addr/id of the submitted event 
+    ["a"/"e", "<addr/id>", <relay>], // addr/id of the submitted event 
+    ["k", "<kind>"], // submitted event's kind, to allow filtering by target event's kind
+    ["r", "<slug>"] // optional slug for submitted event, to be used in the website url, i.e. /posts/<slug>.
+  ]
+}
+```
+
+Only zero or one event may be referenced with `a` or `e` tag.
+
+If `relay` tags are specified in `site` event, `submit` events will only be fetched from those relays.
+
+If target event is authored by none of the contributors, it should be rendered as a `repost` by the contributor and show the original author.
+
+If `r` tag's value starts with `/` then this a relative url of a `static page` (i.e. `r:/path/to/static/page`). The submitted events referenced by `a` or `e` tags of `static pages` SHOULD NOT be included when listing published events, but are only rendered at the specified url. If no `a` or `e` tags are specified in a `submit` event, then it's `content` field will be rendered as the content of the static page - this allows to exclude such static pages from social media feeds.
+
+If several `submit` events published by contributors have the same `r` tag then the most recent one should be preferred.
+
+To override default meta tags of a page, `submit` event may include meta info tags of the `site` event (`title`, `image`, `meta_title`, `og_title` etc). 
+ 
+
+Hashtag pages
+=============
+
+Sites may need to display additional info on web pages dedicated to hashtags, and may use `hashtag` event published by contributors for that. The `hashtag` event is a parameterized replaceable event of kind `30513`. the `content` field may include a string of text in the same format as "long-form content" NIP-23, to be used as the body of the hashtag web page. Hashtag event will have tags:
+
+```
+{
+  "tags":[
+    ["t", "<hashtag>"], // target hashtag, must be included exactly once
+    ["a", "<addr>"], // nostr site address
+    ["r", "<slug>"], // optional slug for hashtag, to be used in the website url, i.e. /tags/<slug>.
+  ]
+}
+```
+
+To override default meta tags of a hashtag page, `hashtag` event may include meta info tags of the `site` event (`title`, `image`, `meta_title`, `og_title` etc). 
+
+If `r` tag starts with `/` then it is a static hashtag webpage, and this hashtag should not be listed under posts and in other hashtag lists.
+
+If several `hashtag` events published by contributors have the same `r` tag then the most recent one should be preferred.
+
+Themes and Plugins
+==================
+
+The `theme` event is parameterized replaceable event of kind `30514`, it's `content` field may have a "rich" description in the same syntax as NIP-23 long-form notes. It has the following tags:
```

Comment:
is there an example content to help understanding how the theme works?

Reply:
Try fetching 30514 events from relay.npubpro.com, and also 1036 code package events linked by themes from the same relay.